scripts/ensure-vhd-images.sh

```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/env bash
+#
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT license.
+
+PUBLISHED_IMAGES=100
+
+exit_code=0
+
+echo "==> Using az CLI to validate published VHD images <=="
+
+az login --service-principal --username "${CLIENT_ID}" --password "${CLIENT_SECRET}" --tenant "${TENANT_ID}" &>/dev/null
+IMAGES=$(az vm image list -p AKS -f aks --all -l eastus | jq '. | length')
```

Comment:
for clarity I suggest renaming `IMAGES` to `ACTUAL_IMAGE_COUNT` and `PUBLISHED_IMAGES` to `EXPECTED_IMAGE_COUNT`